api/envoy/config/metrics/v3/stats.proto

```diff
@@ -330,6 +330,14 @@ message DogStatsdSink {
   // Optional custom metric name prefix. See :ref:`StatsdSink's prefix field
   // <envoy_api_field_config.metrics.v3.StatsdSink.prefix>` for more details.
   string prefix = 3;
+
+  // Optional max datagram size to use when sending UDP messages. By default Envoy
+  // will emit one metric per datagram. By specifying a max-size larger than a single
+  // metric, Envoy will emit multiple, new-line separated metrics. The max datagram
+  // size should not exceed your network's MTU.
+  //
+  // Note that this value may not be respected if smaller than a single metric.
+  uint64 max_bytes_per_datagram = 4;
```

Comment:
This defaults to 0 which while OK is a bit less clear than using the well known type message. Can you switch to that? I wonder if we should just default this to 1400 or something, but I suppose that is a behavior change and might not be worth it.

Reply:
@mattklein123 I'm not too familiar with what this would mean. So if I use `google.protobuf.UInt64Value` here, and also use `absl::optional` (as you mentioned below), does that mean I would need something like:

```proto
bool has_max_bytes_per_datagram = 4;

google.protobuf.UInt64Value max_bytes_per_datagram = 5 [
  (validate.rules).uint64 = {gt: 0}
};
```

Reply:
Remove `bool has_max_bytes_per_datagram = 4;` and just use `google.protobuf.UInt64Value max_bytes_per_datagram = 5` but otherwise yes.

Reply:
Updated.